File: little-bee/controller/router/vsi.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_VSI = "/config/vsi"
const URL_ROUTER_VSI_BY_ID = "/config/vsi/:vsi-id"
const URL_ROUTER_VSI_QUERY = "/config/vsi/query"
const URL_ROUTER_VSI_ALL = "/config/vsi/all"

func VsiRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_VSI, config.CreateVsi)
	r.GET(URL_ROUTER_VSI_BY_ID, config.RetrieveVsi)
	r.PUT(URL_ROUTER_VSI, config.UpdateVsi)
	r.DELETE(URL_ROUTER_VSI_BY_ID, config.DeleteVsi)
	r.PUT(URL_ROUTER_VSI_BY_ID, config.DeleteVsiWithFlag)
	return r
}

File: frontend/console/src/api/vsi.js
import Axios from "@/http";

export function createVsi(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/vsi', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveVsi(vsiId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/vsi/${ vsiId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateVsi(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/vsi', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteVsi(vsiId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/vsi/${ vsiId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteVsiWithFlag(vsiId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/vsi/${ vsiId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllVsis() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/vsi/all').then(res => {
            resolve(res)
        })
    })
}

export function queryVsi (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/vsi/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}